libmultilabel/model.py

```diff
@@ -147,12 +147,21 @@ def test_epoch_end(self, step_outputs):
 
     def _shared_eval_step(self, batch, batch_idx):
         loss, pred_logits = self.shared_step(batch)
-        return {'loss': loss,
+        return {'batch_idx': batch_idx,
+                'loss': loss,
                 'pred_scores': torch.sigmoid(pred_logits),
                 'target': batch['label']}
 
     def _shared_eval_step_end(self, batch_parts):
-        return self.eval_metric.update(batch_parts['pred_scores'], batch_parts['target'])
+        batch_size, num_classes = batch_parts['target'].shape
+        indexes = torch.arange(
+            batch_size*batch_parts['batch_idx'], batch_size*(batch_parts['batch_idx']+1))
+        indexes = indexes.unsqueeze(1).repeat(1, num_classes)
+        return self.eval_metric.update(
+            preds=batch_parts['pred_scores'],
+            target=batch_parts['target'],
+            indexes=indexes
+        )
```

Comment:
not clear about changes from 156-164. Maybe some discussion tomorrow

Reply:
Note: add comments